backend/app/groups/service.py

```
@@ -348,14 +364,25 @@ async def remove_member(self, group_id: str, member_id: str, user_id: str) -> bo
+                status_code=400,
+                detail="Cannot remove member with outstanding balance. Ask them to settle dues first."
+            )
 
         result = await db.groups.update_one(
             {"_id": obj_id},
             {"$pull": {"members": {"userId": member_id}}}
         )
         return result.modified_count == 1
+   
+    async def has_outstanding_balance(self, group_id: str, user_id: str) -> bool:
+        balance_info = await expense_service.get_user_balance_in_group(
```

Ruff (0.12.2)

Undefined name `expense_service`

(F821)